plugin/forward/connect.go

```diff
@@ -26,7 +26,14 @@ func (p *Proxy) readTimeout() time.Duration {
 
 func (p *Proxy) updateRtt(newRtt time.Duration) {
 	rtt := time.Duration(atomic.LoadInt64(&p.avgRtt))
-	atomic.AddInt64(&p.avgRtt, int64((newRtt-rtt)/rttCount))
+	avg := rtt + ((newRtt - rtt) / rttCount)
+	if avg < minTimeout {
```

Comment:
consider moving `if` logic to `readTimeout()` method. It's desirable to execute the two atomic instructions as closely as possible